subsys/net/lib/dns/dns_sd.c

```diff
@@ -575,6 +575,7 @@ int add_srv_record(const struct dns_sd_rec *inst, uint32_t ttl,
 	size_t label_size;
 	uint16_t inst_offs;
 	uint16_t offset = buf_offset;
+	const char *hostname = net_hostname_get();
```

Comment:
This function is already pretty testable as a unit. Is there any reason not to add `hostname` as a function parameter?

This change effectively hard-codes the host name, in that it is not easily manipulated by the test harness that tests `add_srv_record()` as a unit.